charmcraft/commands/pack.py

```diff
@@ -259,150 +240,41 @@ def _pack_bundle(
         # predefined values set automatically.
         bundle_part = config_parts.get("bundle")
         if bundle_part and bundle_part.get("plugin") == "bundle":
-            special_bundle_part = bundle_part
-        else:
-            special_bundle_part = None
-
-        # get the config files
-        bundle_filepath = project.dirpath / "bundle.yaml"
-        bundle_name = bundle_config.get("name")
-        if not bundle_name:
-            raise CraftError(
-                "Invalid bundle config; missing a 'name' field indicating the bundle's name in "
-                "file {!r}.".format(str(bundle_filepath))
-            )
-
-        if special_bundle_part:
             # set prime filters
             for fname in MANDATORY_FILES:
                 fpath = project.dirpath / fname
                 if not fpath.exists():
                     raise CraftError(f"Missing mandatory file: {str(fpath)!r}.")
-            prime = special_bundle_part.setdefault("prime", [])
+            prime = bundle_part.setdefault("prime", [])
             prime.extend(MANDATORY_FILES)
 
             # set source if empty or not declared in charm part
-            if not special_bundle_part.get("source"):
-                special_bundle_part["source"] = str(project.dirpath)
-
-        if env.is_charmcraft_running_in_managed_mode():
-            work_dir = env.get_managed_environment_home_path()
-        else:
-            work_dir = project.dirpath / const.BUILD_DIRNAME
+            if not bundle_part.get("source"):
+                bundle_part["source"] = str(project.dirpath)
 
         # run the parts lifecycle
         emit.debug(f"Parts definition: {config_parts}")
-        lifecycle = parts.PartsLifecycle(
-            config_parts,
-            work_dir=work_dir,
-            project_dir=project.dirpath,
-            project_name=bundle_name,
-            ignore_local_sources=[bundle_name + ".zip"],
-        )
+
         try:
-            lifecycle.run(Step.PRIME)
+            output_files = builder.pack_bundle(
```

Comment:
Maybe we could have the zip name returned separately to avoid giving special meaning to the last element?

Reply:
Merging this conversation with https://github.com/canonical/charmcraft/pull/1230#discussion_r1303513129